### PR TITLE
Add jest configuration for packages which do not have any tests

### PR DESCRIPTION
**Pull request by vzhukovs**
### What does this PR do?
This changes proposal adds jest's configuration for packages which don't have tests.

Signed-off-by: Vladyslav Zhukovskyi <vzhukovs@redhat.com>

### Screenshot/screencast of this PR
N/A

### What issues does this PR fix or reference?
fixes https://github.com/eclipse/che/issues/19130

### How to test this PR?
No need to tests. It's internal stuff, which publishes the coverage report to codecov.io.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### Happy Path Channel

HAPPY_PATH_CHANNEL=stable

### CODE REVIEW

**Comment by vzhukovs** [comment]
[crw-ci-test --rebuild]

File: extensions/eclipse-che-theia-remote-impl-che-server/package.json [hunk 1]
```diff
@@ -18,15 +18,15 @@
...
+    "prepare": "yarn clean && yarn build && yarn test",
     "clean": "rimraf lib",
     "format": "if-env SKIP_FORMAT=true && echo 'skip format check' || prettier --check '{src,tests}/**/*.ts' package.json",
     "format:fix": "prettier --write '{src,tests}/**/*.ts' package.json",
     "lint": "if-env SKIP_LINT=true && echo 'skip lint check' || eslint --cache=true --no-error-on-unmatched-pattern=true '{src,tests}/**/*.ts'",
     "lint:fix": "eslint --fix --cache=true --no-error-on-unmatched-pattern=true \"{src,tests}/**/*.{ts,tsx}\"",
     "compile": "tsc",
     "build": "concurrently -n \"format,lint,compile\" -c \"red,green,blue,purple\" \"yarn format\" \"yarn lint\" \"yarn compile\"",
-    "test": "if-env SKIP_TEST=true && echo 'skip test' || jest --forceExit",
+    "test": "jest",
```
Comment (sunix) [inline extensions/eclipse-che-theia-remote-impl-che-server/package.json:29]: Would it still be possible to skip the tests ?

File: extensions/che-theia-hosted-plugin-manager-extension/package.json [hunk 2]
```diff
@@ -15,20 +15,36 @@
...
       "backend": "lib/node/hosted-plugin-manager-contribution-extension-backend-module"
     }
-  ]
+  ],
+  "jest": {
+    "clearMocks": true,
+    "collectCoverage": true,
+    "collectCoverageFrom": [
+      "src/**/*.{ts,tsx}",
+      "!**/node_modules/**",
```
Comment (benoitf) [inline extensions/che-theia-hosted-plugin-manager-extension/package.json:40]: hello, could you explain why we need to exclude tests or node modules folders ? AFAIK if we include only src folder we never have these

File: extensions/che-theia-hosted-plugin-manager-extension/package.json [hunk 3]
```diff
@@ -15,20 +15,36 @@
     "@eclipse-che/theia-remote-api": "^0.0.1"
   },
   "scripts": {
-    "prepare": "yarn clean && yarn build",
+    "prepare": "yarn clean && yarn build && yarn test",
```
Comment (benoitf) [inline extensions/che-theia-hosted-plugin-manager-extension/package.json:18]: Tests should be skipped if ENV variable SKIP_TEST=true is defined

File: extensions/eclipse-che-theia-remote-impl-che-server/package.json (hunk 1 above)
Comment (benoitf) [inline extensions/eclipse-che-theia-remote-impl-che-server/package.json:29]: ForceExit allows to not have infinite jobs

File: extensions/che-theia-hosted-plugin-manager-extension/package.json (hunk 3 above)
Comment (azatsarynnyy) [inline extensions/che-theia-hosted-plugin-manager-extension/package.json:18]: Why do we need to call `test` within the `prepare` script?

File: extensions/eclipse-che-theia-remote-impl-che-server/package.json (hunk 1 above)
Comment (vzhukovs) [reply extensions/eclipse-che-theia-remote-impl-che-server/package.json]: Fixed

File: extensions/che-theia-hosted-plugin-manager-extension/package.json (hunk 3 above)
Comment (vzhukovs) [reply extensions/che-theia-hosted-plugin-manager-extension/package.json:18]: Added check for the environment variable to all necessary packages

File: extensions/che-theia-hosted-plugin-manager-extension/package.json (hunk 2 above)
Comment (vzhukovs) [reply extensions/che-theia-hosted-plugin-manager-extension/package.json]: It were additional precautions, just to be sure, that we won't get into coverage report any other directories. Removed unnecessary directories from the configs.

File: extensions/eclipse-che-theia-remote-impl-che-server/package.json (hunk 1 above)
Comment (vzhukovs) [reply extensions/eclipse-che-theia-remote-impl-che-server/package.json]: Yep, added the opportunity to skip the tests.

File: extensions/che-theia-hosted-plugin-manager-extension/package.json (hunk 3 above)
Comment (vzhukovs) [reply extensions/che-theia-hosted-plugin-manager-extension/package.json:18]: We can create `"postbuild": "yarn test"` yarn script to run the tests. 

This was aligned to other approaches, we have:

- [recommendations-plugin/package.json#L38](https://github.com/eclipse/che-theia/blob/master/plugins/recommendations-plugin/package.json#L38) 
- [resource-monitor-plugin/package.json#L23](https://github.com/eclipse/che-theia/blob/master/plugins/resource-monitor-plugin/package.json#L23) 
- [workspace-plugin/package.json#L26](https://github.com/eclipse/che-theia/blob/master/plugins/workspace-plugin/package.json#L26) 
- [ports-plugin/package.json#L115](https://github.com/eclipse/che-theia/blob/master/plugins/ports-plugin/package.json#L115)

**Review by azatsarynnyy — APPROVED** [review]

**Review by benoitf — APPROVED** [review]

**Review by sunix — APPROVED** [review]
Thanks :)

**Pull request merged**